lib/std/smartptrs.nim

```diff
@@ -7,12 +7,18 @@
 #    distribution, for details about the copyright.
 
 ## C++11 like smart pointers. They always use the shared allocator.
+## Experimental API, subject to change.
+
+when not compileOption("threads"):
+  {.error: "Smartptrs requires --threads:on option.".}
+
 import std/isolation
 
-template checkNotNil() =
+template checkNotNil(cond: untyped, msg: typed) =
   when compileOption("boundChecks"):
-    if SharedPtr[T](p).val == nil:
-      raise newException(NilAccessDefect, "deferencing nil const pointer")
+    {.line.}:
+      if not cond:
+        raise newException(NilAccessDefect, msg)
```

Comment:
astToStr(condd) & " failed: " & msg

Reply:
Its just p.val != nil why would people need to see that, the error msg is already pretty descriptive.